EssentialFeed/EssentialFeedTests/Feed/Image Comments/API/ImageCommentsMapperTests.swift

```diff
@@ -0,0 +1,88 @@
+//
+// Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import XCTest
+import EssentialFeed
+
+class ImageCommentsMapperTests: XCTestCase {
+	func test_map_throwsErrorOnNon2xxHTTPResponse() throws {
+		let samples = [199, 300, 400, 500]
+
+		try samples.forEach { code in
+			XCTAssertThrowsError(
+				try ImageCommentsMapper.map(anyData(), from: HTTPURLResponse(statusCode: code))
+			)
+		}
+	}
+
+	func test_map_deliversInvalidDataErrorOn2xxHTTPResponseWithEmptyData() throws {
+		let emptyData = Data()
+
+		try successCodes.forEach { code in
+			XCTAssertThrowsError(
+				try ImageCommentsMapper.map(emptyData, from: code)
+			)
+		}
```

Comment:
Following from my previous comment, if you decide to rename `successCodes`, remember to rename the property inside the forEach closure:
```suggestion
		try successfulResponses.forEach { response in
			XCTAssertThrowsError(
				try ImageCommentsMapper.map(emptyData, from: response)
			)
		}
```